src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/AnnotationSetScenarios.java

```diff
@@ -36,147 +41,262 @@
+                .body("documentId", is(newDocumentId.toString()))
+                .log().all();
+    }
 
-        jsonObject.remove("documentId");
-        jsonObject.put("documentId", UUID.randomUUID().toString());
+    @Test
+    public void shouldReturn400WhenUpdateAnnotationSetWithoutId() {
+        final UUID annotationSetId = UUID.randomUUID();
+        final UUID documentId = UUID.randomUUID();
+        final ValidatableResponse response = createAnnotationSet(annotationSetId, documentId);
+        final JSONObject annotationSet = extractJSONObjectFromResponse(response);
+        annotationSet.remove("id");
 
         request
-                .body(jsonObject.toString())
+                .body(annotationSet.toString())
                 .put("/api/annotation-sets")
                 .then()
-                .statusCode(200);
+                .statusCode(400)
+                .log().all();
     }
 
     @Test
-    public void testDeleteAnnotationSetSuccess() {
-        JSONObject jsonObject = new JSONObject();
-        String annotationSetId = UUID.randomUUID().toString();
+    public void shouldReturn401WhenUnAuthenticatedUserUpdateAnnotationSet() {
+        final UUID annotationSetId = UUID.randomUUID();
+        final UUID documentId = UUID.randomUUID();
+        final ValidatableResponse response = createAnnotationSet(annotationSetId, documentId);
+        final JSONObject annotationSet = extractJSONObjectFromResponse(response);
+        final UUID newDocumentId = UUID.randomUUID();
+        annotationSet.put("documentId", newDocumentId);
+
+        unAuthenticatedRequest
+                .body(annotationSet.toString())
+                .put("/api/annotation-sets")
+                .then()
+                .statusCode(401)
+                .log().all();
+    }
 
-        jsonObject.put("documentId", UUID.randomUUID().toString());
-        jsonObject.put("id", annotationSetId);
+    @Test
+    public void shouldReturn200WhenDeleteAnnotationSetById() {
+        final UUID annotationSetId = UUID.randomUUID();
+        final UUID documentId = UUID.randomUUID();
+        final ValidatableResponse response = createAnnotationSet(annotationSetId, documentId);
+        final String id = extractJSONObjectFromResponse(response).getString("id");
+        final ValidatableResponse deletedResponse = deleteAnnotationSetById(id);
+
+        deletedResponse.statusCode(200);
+    }
 
-        request
-                .body(jsonObject.toString())
-                .post("/api/annotation-sets")
-                .then()
-                .statusCode(201);
+    @Test
+    public void shouldReturn500WhenDeleteAnnotationSetByNonExistentId() {
+        final String nonExistentAnnotationSetId = UUID.randomUUID().toString();
+        final ValidatableResponse deletedResponse = deleteAnnotationSetById(nonExistentAnnotationSetId);
 
-        request
-                .delete("/api/annotation-sets/" + annotationSetId)
+        deletedResponse.statusCode(500);
```

Comment:
If the id is non existant...It should be a HTTP status 404...(Resource Not Found).A 500 is an internal server Error.

Reply:
yes, its a bug. Raised a bug for this issue.